sources/packages/backend/apps/api/src/route-controllers/education-program-offering/education-program-offering.institutions.controller.ts

```diff
@@ -76,10 +80,66 @@ export class EducationProgramOfferingInstitutionsController extends BaseControll
     private readonly educationProgramOfferingControllerService: EducationProgramOfferingControllerService,
     private readonly educationProgramOfferingImportCSVService: EducationProgramOfferingImportCSVService,
     private readonly educationProgramOfferingValidationService: EducationProgramOfferingValidationService,
+    private readonly offeringValidationService: EducationProgramOfferingValidationService,
   ) {
     super();
   }
 
+  /**
+   * Validates an offering payload providing the validation result and
+   * study break calculations also used to perform the validation process.
+   * Return a 200 HTTP status instead of 201 to indicate that the operation
+   * was completed with success but no resource was created.
+   * @param locationId location id.
+   * @param programId program id.
+   * @param payload offering data to be validated.
+   * @returns offering validation result.
+   */
+  @HasLocationAccess("locationId")
+  @ApiNotFoundResponse({
+    description: "Not able to find the program in the provided location.",
+  })
+  @Post("location/:locationId/education-program/:programId/validation")
+  @HttpCode(HttpStatus.OK)
+  async validateOffering(
+    @Param("locationId", ParseIntPipe) locationId: number,
+    @Param("programId", ParseIntPipe) programId: number,
+    @Body() payload: EducationProgramOfferingAPIInDTO,
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<OfferingValidationResultAPIOutDTO> {
+    const offeringValidationModel =
+      await this.educationProgramOfferingControllerService.buildOfferingValidationModel(
+        userToken.authorizations.institutionId,
+        locationId,
+        programId,
+        payload,
+      );
+
+    const offeringValidation =
+      this.offeringValidationService.validateOfferingModel(
+        offeringValidationModel,
+        true,
+      );
+
+    const calculatedStudyBreaks =
+      EducationProgramOfferingService.getCalculatedStudyBreaksAndWeeks(
+        offeringValidationModel,
+      );
+
+    return {
+      offeringStatus: offeringValidation.offeringStatus,
+      errors: offeringValidation.errors,
+      warnings: offeringValidation.warnings,
+      studyPeriodBreakdown: {
+        fundedStudyPeriodDays: calculatedStudyBreaks.fundedStudyPeriodDays,
+        totalDays: calculatedStudyBreaks.totalDays,
+        totalFundedWeeks: calculatedStudyBreaks.totalFundedWeeks,
+        unfundedStudyPeriodDays: calculatedStudyBreaks.unfundedStudyPeriodDays,
+      },
+      validationDate: new Date(),
```

Comment:
What is the purpose of validation date here?

Reply:
From an API perspective, it provides the date when the validation was executed.
From a UI perspective, it is used to check if the response is the most updated. This method is called on UI during the onChange of some form.io components, which can lead to possibly more than one call to the method under a poor network connection. The idea was, in case it happens, to update the values on the UI only for the most updated response, discarding any old ones.

Reply:
@ann-aot @guru-aot @andrepestana-aot, as per suggested by @dheepak-aot I am removing the `validationDate` from the API payload and using the HTTP header `last-modified` instead. The API is not adding this header and the Vue application is consuming it.
![image](https://user-images.githubusercontent.com/61259237/200921870-9f0a9d5c-d947-4937-987d-bdeb763078ae.png)